src/main/java/Hachi.java

```diff
@@ -0,0 +1,266 @@
+        System.out.println("    Okay, marking this task as incomplete now:");
+        String statusIcon = listOfTasks[index].getStatusIcon();
+        System.out.print("    " + (index + 1) + ": ");
+        System.out.print("[" + statusIcon + "] ");
+        System.out.println(listOfTasks[index].getName());
+    }
+
+    /**
+     * Prints to the console a goodbye message for the user.
+     */
+
+    public static void printGoodbyeMessage() {
+        System.out.println("    Goodbye! Hope you have a marvelous day.");
+        spacerInsert("medium", true);
+    }
+
+    /**
+     * The main program that starts the chatbot.
+     * Prints to the console for the user to read its messages.
+     * Greets the user and awaits user input.
+     * <p>
+     * Chatbot can:
+     * <p>1. retrieve list of tasks with user input "list"
+     * <p>2. mark or unmark tasks complete with user input "mark #tasknumber"
+     * <p>3. say goodbye to the user with user input "bye" or "goodbye"
+     * <p>4. add a to-do to the list of task with "todo <event name>"
+     * <p>5. add a deadline to the list of task with "deadline <event name> /by <by date>"
+     * <p>6. add an event to the list of task with "event <event name> /from <start date> /to <end date>"
+     * <p>7. retrieve a list of chatbot commands with "help"
+     *
+     * @param args Command line arguments - not used.
+     */
+
+    public static void main(String[] args) {
+        spacerInsert("medium", false);
+        boolean isBye = false;
+        Task[] listOfTasks = new Task[100];
+        printGreetingMessage();
+        printHelpMessage();
+
+        while (!isBye) {
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+            String cleanedInput = line.toUpperCase().trim();
+            int indexOfSpace = cleanedInput.indexOf(" ");
+            String firstWord;
+            if (indexOfSpace == -1) { // check for single-word inputs
+                firstWord = cleanedInput;
+            } else {
+                firstWord = cleanedInput.substring(0, indexOfSpace);
+            }
+
+            if (firstWord.contains("MARK")) {
+                int indexOfTask = cleanedInput.indexOf("MARK") + 4; // find index of task number
+                int taskNumber = Integer.parseInt(cleanedInput.substring(indexOfTask).trim()); // parse string to int
+                if (cleanedInput.contains("UNMARK")) {
+                    unmarkTask(taskNumber - 1, listOfTasks);
+                } else {
+                    markTask(taskNumber - 1, listOfTasks);
+                }
+            } else {
+                switch (firstWord) {
+                case "LIST":
+                    retrieveList(listOfTasks);
+                    break;
+
+                case "TODO":
+                case "EVENT":
+                case "DEADLINE":
+                    TaskType currentTask;
+                    if (cleanedInput.startsWith("TODO")) {
```

Comment:
try your best to avoid having too many nested if/loop statements!

Reply:
maybe to do this, you can add an "&" in the condition lines to avoid nested statements, but still have your coding logic in tact